### PR TITLE
Legitimately handle versions for packages

### PR DESCRIPTION
**Pull request by cmickeyb**
We have not been very good about version updates. By default the version is extracted from the VERSION file but this requires an explicit commit to update.

This change uses the most recent annotated version tag to determine the major and minor numbers and the commit count to determine that patch number. Future versions should be annotatively tagged as 'vM.N.0'.

### CODE REVIEW

File: bin/get_version [hunk 1]
```diff
@@ -14,67 +14,35 @@
...
+    # third case: this is a dirty, committed post tag release
+    elif len(rest) == 3 :
+        assert rest[2] == 'dirty'
+        count = rest[0]
+        commit = rest[1]
+        dirty = rest[2]
+
+    print('{}.{}.{}'.format(major, minor, count))
+except Exception as e :
+    warnings.warn('failed to compute version, using default'
```
Comment (bvavala) [inline bin/get_version:47]:
```suggestion
    warnings.warn('failed to compute version, using default')
```

Comment (cmickeyb) [reply bin/get_version]: hold for now, trying to get the version computed when there is no repository. CI is a problem.

Comment (cmickeyb) [reply bin/get_version]: Fixed.

**Review by marcelamelara — APPROVED** [review]
LGTM, thanks!!

**Review by bvavala — APPROVED** [review]

**Pull request merged**